charger/bender.go

```diff
@@ -98,10 +102,22 @@ func NewBenderCC(uri string, id uint8) (api.Charger, error) {
 		current: 6, // assume min current
 	}
 
+	// check legacy register set
+	if _, err := wb.conn.ReadHoldingRegisters(bendRegChargePointModel, 10); err != nil {
+		wb.legacy = true
+	}
+
 	// check presence of metering
-	b, err := wb.conn.ReadHoldingRegisters(bendRegActivePower, 2)
-	if err == nil && (binary.BigEndian.Uint32(b) != math.MaxUint32) {
-		return decorateBenderCC(wb, wb.currentPower, wb.currents, wb.chargedEnergy, wb.totalEnergy), nil
+	if !wb.legacy {
```

Comment:
Ich würde versuchen die Logik überall gleich zu machen. Entweder if legacy oder if !legacy. Du drehst es in jeder Funktion. Obs funktioniert kann ich nicht beurteilen.

Reply:
Ja, da hast du wohl Recht. Hatte versucht das diff klein zu halten. ;)

Testen kann ich das leider auch überhaupt nicht, da ich bisher niemanden finde der mir eine Webasto Live mit der veralteten Firmware mal extern freigeben kann. Ich kenne nur die Fehlermeldung wegen fehlender Register und versuche diese zu erkennen. Trial&Error...